integration-tests/src/bin/read.rs

```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use libc::{read, write};
+
+fn main() {
+    let mut buf = [0u8; 16];
+    for i in (0..).map(|x| 1 << x) {
```

Comment:
If `i` is `i32`, this will panic on the 33rd iteration. We probably don't want that. We probably don't want a for loop here since `i` or `x` will eventually grow too large and panic for any large enough input.